src/test/java/hudson/plugins/jira/JiraSite2Test.java

```diff
@@ -170,4 +179,137 @@ public void testDeserializeWithoutCredentials() {
             this.password = Secret.fromString(password);
         }
     }
+
+    @Test
+    @WithoutJenkins
+    public void alternativeURLNotNull() {
+        JiraSite site = new JiraSite(nonExistentUrl, exampleOrg,
+            (StandardUsernamePasswordCredentials) null,
+            false, false,
+            null, false, null,
+            null, true);
+        assertNotNull(site.getAlternativeUrl());
+        assertEquals(exampleOrg, site.getAlternativeUrl());
+    }
+
+    @Test
+    @WithoutJenkins
+    public void ensureUrlEndsWithSlash() {
+        JiraSite jiraSite = new JiraSite(nonExistentUrl.toExternalForm());
+        jiraSite.setAlternativeUrl(exampleOrg.toExternalForm());
+        assertTrue(jiraSite.getUrl().toExternalForm().endsWith("/"));
+        assertTrue(jiraSite.getAlternativeUrl().toExternalForm().endsWith("/"));
+        URL url1 = JiraSite.toURL(nonExistentUrl.toExternalForm());
+        URL url2 = JiraSite.toURL(exampleOrg.toExternalForm());
+        assertTrue(url1.toExternalForm().endsWith("/"));
+        assertTrue(url2.toExternalForm().endsWith("/"));
+    }
+
+    @Test
+    @WithoutJenkins
+    public void urlNulls() {
+        JiraSite jiraSite = new JiraSite(nonExistentUrl.toExternalForm());
+        jiraSite.setAlternativeUrl(" ");
+        URL url1 = JiraSite.toURL("");
+        URL url2 = JiraSite.toURL(" ");
+        assertNotNull(jiraSite.getUrl());
+        assertNull(jiraSite.getAlternativeUrl());
+        assertNull(url1);
+        assertNull(url2);
+    }
+
+    @Test
+    @WithoutJenkins
+    public void ensureMainUrlIsMandatory() {
+        thrown.expect(AssertionError.class);
+        thrown.expectMessage("URL cannot be null");
```

Comment:
it's confusing: why do we say it's null while it's obviously not - it's an empty string?

Reply:
I'll change the message to say `URL cannot be emtpy`.
But that's why we use `Util.fixEmptyAndTrim` so we get nulls instead of dealing with empty string values.